framework/src/org/checkerframework/common/value/ValueCheckerUtils.java

```diff
@@ -97,7 +98,8 @@
         } else if (AnnotationUtils.areSameByClass(anno, BottomVal.class)) {
             values = convertBottomVal(anno, castType);
         } else if (AnnotationUtils.areSameByClass(anno, UnknownVal.class)
-                || AnnotationUtils.areSameByClass(anno, ArrayLen.class)) {
+                || AnnotationUtils.areSameByClass(anno, ArrayLen.class)
+                || AnnotationUtils.areSameByClass(anno, IntRange.class)) {
```

Comment:
I overlook this place. Thanks.

This is one of the reasons why, in the first place, I tried to replace each @IntRange that spans 10 or fewer elements by @IntVal, so that we don't have to consider such cases in the implementation.

We had a discussion about this before in here: https://github.com/typetools/checker-framework/pull/994/files/8cd0b6fb5937ace415309a00bb0292cdff3df28a#diff-d095c6faef0654e3d44948ea6958059aR238

Other places that might be affected by NOT converting @IntRange to @IntVal when spans 10 or fewer elements:
1. Two more cases when considering subtype rules: @DoubleVal <= @IntRange and @IntVal <= @IntRange (<= means assign). If we restrict the number of elements of @IntRange to be greater than 10, then we don't have to consider such cases.
2.  Mixed-type expression involves @IntVal and @IntRange that spans 10 or fewer elements. Do we convert @IntVal to @IntRange anyway as you suggested, or do we convert @IntRange back to @IntVal?

I agree that we should not forbid user from writing @IntRange that spans 10 or fewer elements (nor give them warnings for doing so). However, IMO, silently convert @IntRange to @IntVal in such cases might be a good idea for the sake of complexity of implementation. Otherwise, we have to check if @IntRange spans 10 or fewer elements in each and every places, and then do whatever analysis accordingly.